examples/opt/model/test_text_gen.py

```diff
@@ -193,12 +206,13 @@ def inference_func(input_ids, past_key_values, output_attentions=False,
     parser.add_argument("--model", type=str, default="alpa/opt-125m")
     parser.add_argument("--device", type=str, default="cuda")
     parser.add_argument("--dummy", action="store_true")
+    parser.add_argument("--num-beams", type=int, default=1)
     args = parser.parse_args()
 
     tokenizer = GPT2Tokenizer.from_pretrained("facebook/opt-125m")
 
     tic = time.time()
-    model = get_model(args.model, args.device, args.dummy)
+    model = get_model(args.model, args.device, args.dummy, args.num_beams)
```

Comment:
this way of passing `num_beams` seems not aligned with our current interface

Reply:
we must pass a beam here when compile?

Reply:
Beam size determines input tensor's shape and thus the pipeshard executable